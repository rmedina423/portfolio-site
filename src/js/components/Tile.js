import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

export default function Tile(props) {
  return (
    <Link to={props.link} className="tile">
      <div className="tile-overlay">
        <div className="tile-title">
          <h2>{props.title}</h2>
          <h3>— view —</h3>
        </div>
      </div>
      <img className="tile-img" src={`./images/${props.imgName}`} alt={props.title} />
    </Link>
  );
}

Tile.propTypes = {
  link: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  imgName: PropTypes.string.isRequired,
};