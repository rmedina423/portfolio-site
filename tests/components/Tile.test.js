import React from 'react';
import { shallow } from 'enzyme';

import Tile from '../../src/js/components/Tile';

describe('Tile Component', function() {
  beforeEach(() => {
    this.tile = shallow(<Tile
      link="'/another-page"
      title="The Chocolate Room1"
      imgName="ryan_pool_CO3.jpg"
      tileWidth="100px"
      />);
  });

  test('should match snapshot', () => {
    expect(this.tile.getElement()).toMatchSnapshot();
  });
});
