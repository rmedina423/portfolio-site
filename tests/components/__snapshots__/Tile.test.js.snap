// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tile Component should match snapshot 1`] = `
<Link
  className="tile"
  replace={false}
  to="'/another-page"
>
  <div
    className="tile-overlay"
  >
    <div
      className="tile-title"
    >
      <h2>
        The Chocolate Room1
      </h2>
      <h3>
        — view —
      </h3>
    </div>
  </div>
  <img
    alt="The Chocolate Room1"
    className="tile-img"
    src="/photos/ryan_pool_CO3.jpg"
  />
</Link>
`;
